src/util/simplify_expr.cpp

```diff
@@ -656,28 +656,33 @@ simplify_exprt::simplify_typecast(const typecast_exprt &expr)
     config.ansi_c.NULL_is_zero &&
     (expr_type.id() == ID_signedbv || expr_type.id() == ID_unsignedbv) &&
     op_type.id() == ID_pointer && expr.op().id() == ID_plus &&
-    expr.op().operands().size() == 2 &&
-    ((expr.op().op0().id() == ID_typecast &&
-      to_typecast_expr(expr.op().op0()).op().is_zero()) ||
-     (expr.op().op0().is_constant() &&
-      to_constant_expr(expr.op().op0()).get_value() == ID_NULL)))
-  {
-    auto sub_size = pointer_offset_size(to_pointer_type(op_type).subtype(), ns);
-    if(sub_size.has_value())
+    expr.op().operands().size() == 2)
+  {
+    const auto &op_plus_expr = to_plus_expr(expr.op());
+
+    if(((op_plus_expr.op0().id() == ID_typecast &&
```

Comment:
There is still an `op0()` call here; not sure if that is intentional.

Reply:
Yes; note that these are run-time checked in a `multi_ary_exprt`, so this is memory safe.